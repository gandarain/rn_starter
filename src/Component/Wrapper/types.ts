import { ViewStyle } from 'react-native';

export type Props = {
  children: JSX.Element | JSX.Element[] | string | null;
};

export type Styles = {
  container: ViewStyle;
};
