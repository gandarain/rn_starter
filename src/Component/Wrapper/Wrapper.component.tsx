import React from 'react';
import { SafeAreaView, StatusBar } from 'react-native';

import { Colors } from '../../Constant';
import type { Props } from './types';
import config from './Wrapper.config';
import styles from './Wrapper.styles';

/**
 * Wrapper
 * @param {Props} props - props
 * @returns {React.FC<Props>} - Component
 * @constructor
 */
const Wrapper: React.FC<Props> = ({ children }: Props): JSX.Element => (
  <SafeAreaView style={styles.container}>
    <StatusBar animated={true} backgroundColor={Colors.STATUS_BAR} />
    {children}
  </SafeAreaView>
);

Wrapper.displayName = config.displayName;
Wrapper.defaultProps = config.defaultProps;

export default Wrapper;
