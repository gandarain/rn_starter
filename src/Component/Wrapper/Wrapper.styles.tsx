import { StyleSheet } from 'react-native';

import type { Styles } from './types';

export default StyleSheet.create<Styles>({
  container: {
    flex: 1,
    alignItems: 'center',
  },
});
