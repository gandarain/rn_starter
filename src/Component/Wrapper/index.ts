export { default } from './Wrapper.component';
