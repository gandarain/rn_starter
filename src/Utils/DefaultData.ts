/* eslint-disable require-jsdoc */
const emptyString = '';
const emptyObject = {};
const noop = () => {};

export { emptyObject, emptyString, noop };
