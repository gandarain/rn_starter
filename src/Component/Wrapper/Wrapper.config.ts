import Utils from '../../Utils';

const {
  DefaultData: { emptyString },
} = Utils;

const displayName = 'Wrapper';
const defaultProps = {
  children: emptyString,
};

export default { displayName, defaultProps };
