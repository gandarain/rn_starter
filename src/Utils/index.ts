// @flow
import * as DefaultData from './DefaultData';

export default {
  DefaultData,
};
