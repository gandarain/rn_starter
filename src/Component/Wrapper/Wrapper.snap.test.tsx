import * as React from 'react';
import renderer from 'react-test-renderer';

import Utils from '../../Utils';

const {
  DefaultData: { emptyString },
} = Utils;
import Wrapper from './Wrapper.component';

const defaultProps = {
  children: emptyString,
};

describe('<Wrapper />', () => {
  it('Should render Wrapper', () => {
    const tree = renderer.create(<Wrapper {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
