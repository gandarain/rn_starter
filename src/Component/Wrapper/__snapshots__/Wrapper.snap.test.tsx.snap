// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Wrapper /> Should render Wrapper 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  
</RCTSafeAreaView>
`;
